### PR TITLE
(fix): fix all scores zip.

### PR DESCRIPTION
**Pull request by PanAndy**
A maximum of 5 rounds, a minimum of 1 round
```
[len(s) for s in all_scores]
```
[3, 4, 3, 5, 4, 5, 2, 1]

The purpose is to place the reward of each round of interaction at the corresponding position
zip is based on the shortest list in *all_scores, len(list(zip(*all_scores))) = 1
```
for idx, scores in enumerate(list(zip(*all_scores))):
    scores = torch.tensor(scores, dtype=torch.float32)
    turn_indicator = idx * 2 + 3 # 0: pad. 1: system. 2+2n: user. 3+2n: assistant
    reward_position = (input_ids == reward_token) & (turn_indicators == turn_indicator)
    score_tensor[reward_position] = scores
```

So here we should use zip_longest(*all_scores, fillvalue=0), len(list(zip_longest(*all_scores, fillvalue=0))) = 5
```
for idx, scores in enumerate(zip_longest(*all_scores, fillvalue=0)):
    pass
```

### CODE REVIEW

**Comment by ZihanWang314** [comment]
Thanks for the PR! Merged.
